components/RoomList/RoomList.js

```diff
@@ -1,22 +1,37 @@
-import styled from "styled-components";
-import RoomTile, { RoomType } from "@/components/RoomTile/RoomTile.js";
+import useSWR from "swr";
+import { fetcher } from "@/utils/fetcher";
+import RoomTile from "@/components/RoomTile/RoomTile.js";
+import {
+  StyledError,
+  StyledErrorH3,
+  StyledLoading,
+  GridContainer,
+} from "./RoomsList.Styled";
+import StyledLink from "../StyledLink/StyledLink";
 
-// Styled Component
-const GridContainer = styled.div`
-  display: grid;
-  grid-template-columns: repeat(2, 1fr);
-  justify-items: center;
-  gap: 20px;
-  margin: 0px 40px 100px 40px;
-`;
+export default function RoomList() {
+  const { data: rooms, error, isLoading } = useSWR("/api/rooms", fetcher);
 
-// List of Rooms (mapped over dummy array which represents user rooms from e.g. db/localstroage )
-export default function RoomList({ rooms }) {
+  if (error) {
+    return (
+      <StyledError>
+        <StyledErrorH3>ERROR</StyledErrorH3>Failed to load rooms ☹︎
+      </StyledError>
+    );
+  }
+
+  if (isLoading) {
+    return <StyledLoading>Loading your rooms...</StyledLoading>;
+  }
+
+  // TODO: empty state
   return (
     <>
       <GridContainer>
         {rooms.map((room, index) => (
-          <RoomTile key={index} title={room.title} type={room.type} />
+          <StyledLink key={index} href={`/rooms/${room._id}`}>
+            <RoomTile key={index} title={room.title} type={room.type} />
```

Comment:
Not sure if this is intended. But you don't need the `key` prop here. But again, not sure if you need it in the `RoomTile` component.